crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -29,11 +29,17 @@ pub use self::serde::*;
 pub use map_entities::*;
 
 use crate::{archetype::ArchetypeId, storage::SparseSetIndex};
-use std::{
-    convert::TryFrom,
-    fmt, mem,
-    sync::atomic::{AtomicI64, Ordering},
-};
+use std::{convert::TryFrom, fmt, mem, sync::atomic::Ordering};
+
+#[cfg(target_has_atomic = "64")]
+use std::sync::atomic::AtomicI64 as AtomicInt;
```

Comment:
This should have a comment for posterity, mentioning that all mainstream platforms do make 64 bit atomics available, but this technique adds support for them with some limitations.